packages/syft/src/syft/service/code/user_code.py

```diff
@@ -609,29 +649,25 @@ def show_code_cell(self):
         ip.set_next_input(warning_message + self.raw_code)
 
 
-@migrate(UserCode, UserCodeV1)
-def downgrade_usercode_v2_to_v1():
+@migrate(UserCode, UserCodeV2)
+def downgrade_usercode_v3_to_v2():
     return [
-        drop("uses_domain"),
-        drop("nested_requests"),
-        drop("nested_codes"),
+        drop("worker_pool_id"),
     ]
 
 
-@migrate(UserCodeV1, UserCode)
-def upgrade_usercode_v1_to_v2():
+@migrate(UserCodeV2, UserCode)
+def upgrade_usercode_v2_to_v3():
     return [
-        make_set_default("uses_domain", False),
-        make_set_default("nested_requests", {}),
-        make_set_default("nested_codes", {}),
+        make_set_default("worker_pool_id", None),
     ]
 
 
 @serializable(without=["local_function"])
 class SubmitUserCode(SyftObject):
```

Comment:
Q: Since we are also changing SubmitUserCode, do we not need to Migration here, by creation SubmitUserCodeV2?

Reply:
Fixed in [2ca8512](https://github.com/OpenMined/PySyft/pull/8376/commits/2ca8512f8d3b8e5884876ed824a4dc437317fec1)